watsononlinestore/watson_online_store.py

```diff
@@ -180,6 +191,93 @@ def setup_conversation_workspace(conversation_client, environ):
+        # Discovery collection names
+        AMAZON_COLLECTION = "amazon-shopping"
+        IBM_COLLECTION = "ibm-logo-store"
+        # Alternate discovery collection names if we must re-create
+        AMAZON_ALT_COLLECTION = "amazon-shopping-alt"
+        IBM_ALT_COLLECTION = "ibm-logo-store-alt"
+        # File path location to discovery html files
+        AMAZON_DATA_PATH = "data/amazon_data_html/"
+        IBM_DATA_PATH = "data/ibm_store_html/"
+
+        # Ensure we a valid environment exists.
+        try:
+            discovery_client.get_environment(environment_id)
+        except Exception:
+            raise Exception("DISCOVERY_ENVIRONMENT_ID=%s is specified in "
+                            "a runtime environment variable, but that "
+                            "environment does not exist." % environment_id)
+
+        # Determine if collection exists.
+        try:
+            collection = discovery_client.get_collection(environment_id,
+                                                         collection_id)
+            if collection:
+                return collection_id
+        except Exception as e:
+            LOG.debug("Watson Discovery - unable to find collection. "
+                      "Error: %s" % repr(e))
+
+        # Try to find collection by name.
+        for coll in \
+                discovery_client.list_collections(
+                    environment_id).get('collections', []):
+            if ((data_source == AMAZON and
+                    coll['name'] == AMAZON_COLLECTION) or
+                    (data_source == IBM and
+                        coll['name'] == IBM_COLLECTION)):
+                return coll['collection_id']
+
+        # Doesn't exist, so create it.
+        try:
+            if data_source == AMAZON:
+                name = AMAZON_ALT_COLLECTION
+                path = AMAZON_DATA_PATH
+            elif data_source == IBM:
+                name = IBM_ALT_COLLECTION
+                path = IBM_DATA_PATH
+            if name:
+                collection = discovery_client.create_collection(environment_id,
+                                                                name)
+
+            # Add documents to collection
+            if collection:
+                collection_id = collection['collection_id']
+                for root, dirs, files in os.walk(path):
+                    for file in files:
+                        if file.endswith('.html'):
+                            with open(path + file, 'r') as f:
+                                data = f.read()
+                            discovery_client.add_document(environment_id,
+                                                          collection_id,
+                                                          file_data=data)
+
+        except Exception as e:
+            raise Exception("Discovery Collection could not be created. "
+                            "Error: " % repr(e))
+
+        return collection_id
```

Comment:
I guess this could be uninitialized. Probably the else I mentioned earlier should be added to make sure that we can't get to the end w/o one.